app/src/main/java/io/homeassistant/companion/android/sensors/StorageSensorManager.kt

```diff
@@ -113,29 +104,19 @@ class StorageSensorManager : SensorManager {
         if (!isEnabled(context, externalStorage.id))
             return
 
-        externalPath = externalMemoryAvailable(context)
-        var totalExternalStorage = "No SD Card"
-        var freeExternalStorage = "No SD Card"
-        var percentFreeExternal = 0
-
-        if (externalPath != null) {
-            val statSD = StatFs(externalPath.toString())
-            blockSizeSD = statSD.blockSizeLong
-            availableBlocksSD = statSD.availableBlocksLong
-            totalBlocksSD = statSD.blockCountLong
-            totalExternalStorage = getTotalExternalMemorySize()
-            freeExternalStorage = getAvailableExternalMemorySize()
-            percentFreeExternal = ((availableBlocksSD.toDouble() / totalBlocksSD.toDouble()) * 100).roundToInt()
+        val externalStoragePath = getExternalStoragePathIfAvailable(context)
+        val externalStorageStats = externalStoragePath?.let {
+            getStorageStats(it)
         }
 
         onSensorUpdated(
             context,
             externalStorage,
-            percentFreeExternal,
+            externalStorageStats?.percentage ?: 0,
             externalStorage.statelessIcon,
             mapOf(
-                "free_external_storage" to freeExternalStorage,
-                "total_external_storage" to totalExternalStorage
+                "free_external_storage" to (externalStorageStats?.freeBytes ?: "No SD Card"),
+                "total_external_storage" to (externalStorageStats?.totalBytes ?: "No SD Card")

```

Comment:
Off-topic:
These attribute names don't match the ones written in the [documentation](https://companion.home-assistant.io/docs/core/sensors/#storage-sensor)

Reply:
We should update the documentation then, I thought we had updated this a while back but may have missed the documentation. Attribute names should be all lower case with underscores for best HA core compatibility. There may or may not be other places to correct now :)